src/group/tests/kat_messages.rs

```diff
@@ -0,0 +1,512 @@
+    mls_plaintext_commit: String,      /* serialized MLSPlaintext(Commit) */
+    mls_ciphertext: String,            /* serialized MLSCiphertext */
+}
+
+#[cfg(any(feature = "expose-test-vectors", test))]
+pub fn generate_test_vector(ciphersuite: &'static Ciphersuite) -> MessagesTestVector {
+    let ciphersuite_name = ciphersuite.name();
+    let credential_bundle = CredentialBundle::new(
+        b"OpenMLS rocks".to_vec(),
+        CredentialType::Basic,
+        SignatureScheme::from(ciphersuite_name),
+    )
+    .unwrap();
+    let key_package_bundle =
+        KeyPackageBundle::new(&[ciphersuite_name], &credential_bundle, Vec::new()).unwrap();
+    let capabilities = CapabilitiesExtension::default();
+    let lifetime = LifetimeExtension::default();
+
+    // Let's create a group
+    let group_id = GroupId::random();
+    let config = GroupConfig::default();
+    let mut group = MlsGroup::new(
+        &group_id.as_slice(),
+        ciphersuite_name,
+        key_package_bundle,
+        config,
+        None,
+        ProtocolVersion::default(),
+    )
+    .unwrap();
+
+    let ratchet_tree = group.tree().public_key_tree_copy();
+
+    // We can't easily get a "natural" GroupInfo, so we just create one here.
+    let group_info = GroupInfo::new(
+        group_id.clone(),
+        GroupEpoch(0),
+        get_random_vec(ciphersuite.hash_length()),
+        get_random_vec(ciphersuite.hash_length()),
+        vec![Box::new(RatchetTreeExtension::new(ratchet_tree.clone()))],
+        ConfirmationTag(Mac {
+            mac_value: get_random_vec(ciphersuite.hash_length()),
+        }),
+        LeafIndex::from(0u32),
+    );
+    let group_secrets = GroupSecrets::random_encoded(ciphersuite, ProtocolVersion::default());
+    let public_group_state = group.export_public_group_state(&credential_bundle).unwrap();
+
+    // Create some proposals
+    let key_package_bundle =
+        KeyPackageBundle::new(&[ciphersuite_name], &credential_bundle, Vec::new()).unwrap();
+    let key_package = key_package_bundle.key_package();
+
+    let add_proposal = AddProposal {
+        key_package: key_package.clone(),
+    };
+    let update_proposal = UpdateProposal {
+        key_package: key_package.clone(),
+    };
+    let remove_proposal = RemoveProposal {
+        removed: random_u32(),
+    };
+
+    let psk_id = PreSharedKeyId::new(
+        PskType::External,
+        Psk::External(ExternalPsk::new(get_random_vec(ciphersuite.hash_length()))),
+        get_random_vec(ciphersuite.hash_length()),
+    );
+
+    let psk_proposal = PreSharedKeyProposal { psk: psk_id };
+    let reinit_proposal = ReInitProposal {
+        group_id,
+        version: ProtocolVersion::Mls10,
+        ciphersuite: ciphersuite_name,
+        extensions: vec![Box::new(RatchetTreeExtension::new(ratchet_tree.clone()))],
+    };
+    // We don't support external init proposals yet.
+    let external_init_proposal = vec![0u8];
```

Comment:
We should leave the fields we don't support empty instead of adding values. This could be interpreted as a wrong value.